src/jrd/trace/TraceManager.cpp

```diff
@@ -366,31 +325,36 @@ void TraceManager::update_session(const TraceSession& session)
 		}
 	}
 
-	ReadLockGuard guard(init_factories_lock, FB_FUNCTION);
-	if (!factories)
-		return;
+	TraceInitInfoImpl attachInfo(session, attachment, filename);
 
-	for (FactoryInfo* info = factories->begin(); info != factories->end(); ++info)
+	GetPlugins<ITraceFactory> traceItr(IPluginManager::TYPE_TRACE, session.getPluginsString());
+	for (; traceItr.hasData(); traceItr.next())
```

Comment:
This plain-C style syntax (for-loop iterator declared before the loop) makes sense only when iterator is needed after the end of loop. Why not:
```
for (GetPlugins<ITraceFactory> traceItr(IPluginManager::TYPE_TRACE, session.getPluginsString());
	traceItr.hasData(); traceItr.next())
```